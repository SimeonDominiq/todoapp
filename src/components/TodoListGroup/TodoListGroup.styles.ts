import styled from "styled-components"

const TodoListGroupWrapper = styled.div`
  margin-bottom: 20px;

  .group-header {
    display: flex;
    gap: 8px;
    align-items: center;

    .group-sn {
      width: 25px;
      height: 25px;
      display: flex;
      align-items: center;
      justify-content: center;
      background: #000;
      color: #fff;
      border-radius: 50%;
    }

    .group-title {
      font-weight: 600;
      font-size: 1.4rem;
      width: 80%;
    }

    svg {
      height: 25px;
    }
  }
`

export default TodoListGroupWrapper
