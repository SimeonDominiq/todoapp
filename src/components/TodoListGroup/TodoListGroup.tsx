import { FC } from "react"
import { ITodoGroup } from "src/types"
import CheckMark from "../Icons/CheckMark"
import TodoListItem from "../TodoListItem/TodoListItem"
import TodoListGroupWrapper from "./TodoListGroup.styles"

type Props = {
  todoGroup: ITodoGroup
}

const TodoListGroup: FC<Props> = ({ todoGroup }) => {
  const { groupId, groupTitle, todoItems } = todoGroup

  const isGroupCompleted = todoItems.every((item) => item.isCompleted === true)

  return (
    <TodoListGroupWrapper>
      <div className="group-header">
        <span className="group-sn">{groupId}</span>
        <h2 className="group-title">{groupTitle}</h2>
        {isGroupCompleted && <CheckMark />}
      </div>
      {todoItems.map((item) => (
        <TodoListItem key={item.id} {...{ ...item, groupId }} />
      ))}
    </TodoListGroupWrapper>
  )
}

export default TodoListGroup
