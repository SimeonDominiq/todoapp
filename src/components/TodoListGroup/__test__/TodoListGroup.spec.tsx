import renderer from "react-test-renderer"
import { Provider } from "react-redux"
import configureStore from "redux-mock-store"
import TodoListGroup from "../TodoListGroup"

const props = {
  groupId: 1,
  groupTitle: "Foundation",
  todoItems: [
    {
      id: 11,
      title: "Setup virtual office",
      isCompleted: false,
    },
  ],
}

const mockStore = configureStore()

describe("App: Components: Todo List Group Layout", () => {
  let wrapper: any

  it("defines the component", () => {
    wrapper = renderer.create(
      <Provider store={mockStore()}>
        <TodoListGroup todoGroup={props} />
      </Provider>,
    )

    expect(wrapper).toBeDefined()
  })

  it("matches snapshot", () => {
    const output = wrapper.toJSON()
    expect(output).toMatchSnapshot()
  })
})
