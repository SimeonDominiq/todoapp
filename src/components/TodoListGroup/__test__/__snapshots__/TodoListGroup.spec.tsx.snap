// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: Components: Todo List Group Layout matches snapshot 1`] = `
<div
  className="sc-gsnTZi kJiiun"
>
  <div
    className="group-header"
  >
    <span
      className="group-sn"
    >
      1
    </span>
    <h2
      className="group-title"
    >
      Foundation
    </h2>
  </div>
  <div
    className="sc-bczRLJ kLtGVI"
  >
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
        value={11}
      />
       
      Setup virtual office
    </label>
  </div>
</div>
`;
